graphblas/io.py

```diff
@@ -319,6 +375,120 @@ def to_scipy_sparse(A, format="csr"):
 
 
+def to_awkward(A, format=None):
+    """Create an Awkward Array from a GraphBLAS Matrix
+
+    Parameters
+    ----------
+    A : Matrix or Vector
+        GraphBLAS object to be converted
+    format : str {'csr', 'csc', 'hypercsr', 'hypercsc', 'vec}
+        Default format is csr for Matrix; vec for Vector
+
+    The Awkward Array will have top-level attributes based on format:
+    - vec/csr/csc: values, indices
+    - hypercsr/hypercsc: values, indices, offset_labels
+
+    Top-level parameters will also be set: format, shape
+
+    Returns
+    -------
+    awkward.Array
+
+    """
+    import awkward._v2 as ak
+    from awkward._v2.forms.listoffsetform import ListOffsetForm
+    from awkward._v2.forms.numpyform import NumpyForm
+    from awkward._v2.forms.recordform import RecordForm
+
+    out_type = _output_type(A)
+    if format is None:
+        format = "vec" if out_type is _Vector else "csr"
+    format = format.lower()
+    classname = None
+
+    if out_type is _Vector:
+        if format != "vec":
+            raise ValueError(f"Invalid format for Vector: {format}")
+        size = A.nvals
+        indices, values = A.to_values()
+        form = RecordForm(
+            contents=[
+                NumpyForm(A.dtype.numba_type.name, form_key="node1"),
+                NumpyForm("int64", form_key="node0"),
+            ],
+            fields=["values", "indices"],
+        )
+        d = {"node0-data": indices, "node1-data": values}
+
+    elif out_type is _Matrix:
+        if _backend != "suitesparse":
+            raise NotImplementedError(
+                f"Conversion of Matrix to Awkward Array not supported for backend '{_backend}'"
+            )
+        if format not in {"csr", "csc", "hypercsr", "hypercsc"}:
+            raise ValueError(f"Invalid format for Matrix: {format}")
+        if format[-1] == "r":
+            size = A.nrows
+            indices = "col_indices"
+            labels = "rows"
+        else:
+            size = A.ncols
+            indices = "row_indices"
+            labels = "cols"
+        info = A.ss.export(format, sort=True)
+        if info["is_iso"]:
+            info["values"] = _np.ascontiguousarray(_np.broadcast_to(info["values"], A.nvals))
+        form = ListOffsetForm(
+            "i64",
+            RecordForm(
+                contents=[
+                    NumpyForm("int64", form_key="node3"),
+                    NumpyForm(A.dtype.numba_type.name, form_key="node4"),
+                ],
+                fields=["indices", "values"],
+            ),
+            form_key="node1",
+        )
+        d = {
+            "node1-offsets": info["indptr"],
+            "node3-data": info[indices],
+            "node4-data": info["values"],
+        }
+        if format.startswith("hyper"):
+            # Define behaviors to make all fields function at the top-level
+            @ak.behaviors.mixins.mixin_class(ak.behavior)
+            class DoublyCompressedMatrix:
```

Comment:
Want to make this a global that is only defined once?  e.g.
```python

_DoublyCompressedMatrix = None

def to_awkward(A, format=None):
    ...
    global _DoublyCompressedMatrix
    if _DoublyCompressedMatrix is None:
        @ak.behaviors.mixins.mixin_class(ak.behavior)
        class _DoublyCompressedMatrix:
            ...
```

Reply:
Good call. I updated to make the class global.